app/controllers/api/policies_controller.rb

```diff
@@ -3,5 +3,65 @@ class PoliciesController < BaseController
     include Subcollections::Conditions
     include Subcollections::Events
     include Subcollections::PolicyActions
+    REQUIRED_FIELDS = %w(name description towhat conditions_ids action_id event_id).freeze
+
+    def create_resource(type, _id, data = {})
+      assert_id_not_specified(data, type)
+      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
+      create_policy(data)
+    end
+
+    def edit_resource(type, id = nil, data = {})
+      raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
+      policy = resource_search(id, type, collection_class(:policies))
+      add_associations(data, policy)
+      policy.update_attributes(data)
+      policy
+    end
+
+    def delete_resource(type, id = nil, _data = nil)
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
+      policy = resource_search(id, type, MiqPolicy)
+      api_log_info("Deleting #{policy_ident(policy)}")
+      policy.destroy
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    private
+
+    def create_policy(data)
+      policy = MiqPolicy.create!(:name        => data.delete("name"),
+                                 :description => data.delete("description"),
+                                 :towhat      => data.delete("towhat")
+                                )
+      add_associations(data, policy)
+      policy
+    rescue => err
+      policy.destroy if policy
+      raise BadRequestError, "Could not create the new policy - #{err}"
+    end
+
+    def add_associations(data, policy)
+      miq_policy_contents = create_policy_content(data.delete("action_id"), data.delete("event_id"))
+      if miq_policy_contents
+        policy.miq_policy_contents.destroy_all
+        policy.miq_policy_contents << miq_policy_contents
+      end
+      policy.conditions = Condition.where(:id => data.delete("conditions_ids")) if data["conditions_ids"]
+    end
+
+    def create_policy_content(action_id, event_id)
+      if action_id && event_id
+        policy_content = MiqPolicyContent.new
+        policy_content.miq_action = MiqAction.find(action_id)
+        policy_content.miq_event_definition = MiqEventDefinition.find(event_id)
+        policy_content
```

Comment:
it feels like this should be in the model.  Maybe leverage MiqPolicy.add_event ?